tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -87,7 +87,7 @@ symbolMatchingInstance language name str = do
   tsSymbol <- runIO $ withCString str (TS.ts_language_symbol_for_name language)
   tsSymbolType <- toEnum <$> runIO (TS.ts_language_symbol_type language tsSymbol)
   [d|instance TS.SymbolMatching $(conT name) where
-      showFailure _ node = "Expected " <> $(litE (stringL (show name))) <> " but got " <> show (TS.fromTSSymbol (nodeSymbol node) :: $(conT (mkName "Grammar.Grammar")))
+      showFailure _ node = "Expected " <> $(litE (stringL (TS.symbolToName tsSymbolType str))) <> " but got " <> show (TS.fromTSSymbol (nodeSymbol node) :: $(conT (mkName "Grammar.Grammar")))
```

Comment:
There are _two_ of all the language constructs we're trying to parse (`yeild` is `AnonymousYield` the anonymous leaf type and also `Yield` the product type). We're for some reason expecting the leaf when we actually should expect the product, causing us to fail.